### PR TITLE
Minor fixes in Semgrep Pro Engine docs, update screenshots in Findings page, update Demo Project, remove demo project from Semgrep Pro Engine docs

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [ ] Redirects are added if the PR changes page URLs

### CODE REVIEW

File: docs/semgrep-app/findings.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Object.entries(frontMatter).filter(
 
 # Managing findings in Semgrep Cloud Platform
 
-A **finding** is the core result of Semgrep analysis. Findings are generated when a Semgrep rule matches a piece of code.
+Finding page enables you to easily manage **findings**. **Findings** are the core results of Semgrep analysis. The findings is generated when a Semgrep rule matches a piece of code.
```
Comment (adamkvitek) [inline docs/semgrep-app/findings.md:28]:
```suggestion
Finding page enables you to easily manage **findings**. **Findings** are the core results of Semgrep analysis. The findings are generated when a Semgrep rule matches a piece of code.
```

File: docs/deepsemgrep/deepsemgrep-introduction.md [hunk 2]
```diff
@@ -68,31 +68,22 @@ To enable Semgrep Pro Engine in Semgrep Cloud Platform, follow these steps:
...
-
-### Testing through a demo project
-
-After enabling Semgrep Pro Engine, you can add a demo project to quickly try Semgrep Pro Engine.
-
-<AddDemoProject />
-
-You have now successfully run Semgrep Pro Engine and Pro rules on a project.
+1. Ensure that you have the **default ruleset** enabled in your **[Rule Board](https://semgrep.dev/orgs/-/board)**. In the Rule Board, the **default ruleset** is in the **Monitor column**. If this ruleset is **not** added, go to [https://semgrep.dev/p/default](https://semgrep.dev/p/default), and then click **Add to Rule Board**.
+1. Optional: If you don't have any projects added in your organization, follow the procedures described in [Scanning a repository from GitHub or GitLab](/semgrep-code/getting-started-with-semgrep-code/#semgrep-code-with-semgrep-cloud-platform) to scan a new project with Semgrep Pro Engine.
```
Comment (s-santillan) [inline docs/deepsemgrep/deepsemgrep-introduction.md:72]:
```suggestion
1. Optional: If you don't have any projects added in your organization, follow the procedures described in [Scanning a repository from GitHub or GitLab](/semgrep-code/getting-started-with-semgrep-code/#semgrep-code-with-semgrep-cloud-platform) to scan a new project with Semgrep Pro Engine. Ensure that your project's language is supported by Semgrep Pro Engine.
```
Note that the projects they onboard have to be of any language supported by Semgrep Pro Engine. The reason why I used juice shop was because it's used in the CLI for Semgrep Pro.

**Review by milanwilliams — APPROVED** [review]

File: docs/deepsemgrep/deepsemgrep-introduction.md [hunk 3]
```diff
@@ -68,31 +68,22 @@ To enable Semgrep Pro Engine in Semgrep Cloud Platform, follow these steps:
...
 
 ## Additional information
 
 ### Experimental support for interprocedural analysis
 
-Interprocedural analysis keeps the fast scan times of Semgrep OSS while also finding new vulnerabilities that cross functions. It's designed to improve results shown in pull requests and help security engineers build trust with developers.
+Interprocedural analysis keeps the fast scan times of Semgrep OSS while also finding new vulnerabilities that cross functions. It's designed to improve results of pull or merge requests (PRs or MRs) and help security engineers build trust with developers.
```
Comment (s-santillan) [inline docs/deepsemgrep/deepsemgrep-introduction.md:78]: I would revert the change. "shown in" is correct. The result of the semgrep scan is displayed to the users. NOT the result of a pull or merge request.

File: docs/deepsemgrep/deepsemgrep-introduction.md [hunk 4]
```diff
@@ -109,7 +100,7 @@ The following performance differences may arise:
 
 ### Semgrep Pro Engine CI scan issues
 
-To ensure scans reliably complete, Semgrep Pro Engine can fall back to the use of Semgrep OSS Engine. This ensures that in the vast majority of cases, scans run successfully.
+To provide reliability, Semgrep Pro Engine can fall back to the use of Semgrep OSS Engine. This ensures that in the vast majority of cases, scans run successfully.
```
Comment (s-santillan) [inline docs/deepsemgrep/deepsemgrep-introduction.md:103]: The important bit here is that the scan is **completed**. That is more concrete than "reliable" which could mean anything.

Comment (adamkvitek) [reply docs/deepsemgrep/deepsemgrep-introduction.md:103]:
```suggestion
To provide realibly completed scans, Semgrep Pro Engine can fall back to the use of Semgrep OSS Engine. This ensures that in the vast majority of cases, scans run successfully.
```

File: docs/deepsemgrep/deepsemgrep-introduction.md (hunk 3 above)
Comment (adamkvitek) [reply docs/deepsemgrep/deepsemgrep-introduction.md:78]:
```suggestion
Interprocedural analysis keeps the fast scan times of Semgrep OSS while also finding new vulnerabilities that cross functions. It's designed to improve results shown in pull or merge requests (PRs or MRs) and help security engineers build trust with developers.
```

File: docs/semgrep-app/findings.md [hunk 5]
```diff
@@ -133,18 +133,31 @@ Display and filter for findings reported only in specific time. To display findi
 
 ### Grouping by rule
 
-Semgrep Cloud Platform enables you to group findings by a specific rule on the Findings page.
+Semgrep Cloud Platform by default groups findings by a specific rule on the Findings page. Procedure documented in this section helps you to enable group by rule when it was disabled.
```
Comment (s-santillan) [inline docs/semgrep-app/findings.md:136]: Since it's on by default, how about it just ends on:

```suggestion
Semgrep Cloud Platform by default groups findings by a specific rule on the Findings page. 
```

Remove the procedure to group findings by rules and just indicate that they can change back, because it's pointless to have 2 separate procedures just pressing a toggle.

**Review by s-santillan — APPROVED** [review]

File: docs/deepsemgrep/deepsemgrep-introduction.md (hunk 2 above)
Comment (adamkvitek) [reply docs/deepsemgrep/deepsemgrep-introduction.md:72]:
```suggestion
1. Optional: If you don't have any projects added to your organization, follow the procedures described in [Scanning a repository from GitHub or GitLab](/semgrep-code/getting-started-with-semgrep-code/#semgrep-code-with-semgrep-cloud-platform) to scan a new project with Semgrep Pro Engine. Ensure that your project's language is supported by Semgrep Pro Engine.

:::info Testing Semgrep Pro Engine
To test Semgrep Pro Engine on a purposefully vulnerable repository, fork [juice-shop](https://github.com/juice-shop/juice-shop) repository, and then add it to SCP by following the steps described in  [Adding a repository](/semgrep-app/getting-started-with-semgrep-app/#option-b-adding-a-repository-from-github-or-gitlab).
:::
```
